internal/app/connectconformance/results.go

```diff
@@ -413,13 +413,31 @@ func checkError(expected, actual *conformancev1.Error) multiErrors {
 	if len(actual.Details) < length {
 		length = len(actual.Details)
 	}
+	actualPayload := &conformancev1.ConformancePayload{}
+	expectedPayload := &conformancev1.ConformancePayload{}
 	for i := 0; i < length; i++ {
 		// TODO: Should this be more lenient? Are we okay with details getting re-ordered?
 		//       An alternative might be to create a map keyed by type, and for each type
 		//       remove expected messages as they are matched against actual ones.
```

Comment:
I'm actually wondering whether we care about anything at all other than `ConformancePayload`? From a testing standpoint, we really only care that the server can support error details and the client can read what the server sent. Setting the `ConformancePayload` into error details for unary errors should actually cover that case maybe?

Reply:
We also care about how unprocessable messages in an `Any` are handled, like if the server gets an `Any` it doesn't understand from a remote server and then must include it in the error details. This mainly impacts the Connect protocol, since it tries to parse the `Any` to produce the debug info.

Also, I'm not sure it's `ConformancePayload` we care about. I think it's only `RequestInfo`. The `ConformancePayload` also includes a `data` field, that is only used with real responses (when the response definition indicates the data it should return).

Reply:
Yeah I actually started out with just `RequestInfo`, but it was easier to do the `ConformancePayload` because the `checkPayloads` function already existed. But, I split that out into a `checkRequestInfo` function and went with `RequestInfo` instead.